### PR TITLE
Tweak rng tests to support Random123-1.13.2 with LLVM 10.

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

* Recent updates allow Draco to work correctly with Random123 version 1.13.2. Unfortunately, those updates weren't tested with LLVM 10.  This PR makes a few more adjustments to fix the remaining issues.

### Description of changes

+ Exclude ut_gsl.c when compiling with LLVM or MSVC. The file does not conform to the C++ standard.
+ Change file extension on some files to get correct C++ linkage options.
+ Fix unsupported CPP macros.
+ Comment block fixes.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

**Comment by KineticTheory** [comment]
@alexrlongne Here is another round of fixes to support Random123 1.13.2.  This time, this fixes are required for LLVM-10 and are related to C vs C++ allowable code.

File: src/rng/test/time_initkeyctr.h [hunk 1]
```diff
@@ -29,8 +29,8 @@ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
-#ifndef TIME_INITKEYCTR_H__
```
Comment (alexrlongne) [inline src/rng/test/time_initkeyctr.h:32]: I know the leading double underscore is used internally, does LLVM also warn (or throw an error) with the trailing double underscores?

**Review by alexrlongne — APPROVED** [review]

File: src/rng/test/time_initkeyctr.h (hunk 1 above)
Comment (KineticTheory) [reply src/rng/test/time_initkeyctr.h:32]: Yeah.  Newer versions of LLVM are very picky about this stuff. "violates C++ standard and may lead to undefined behavior" or similar warning.

Comment (alexrlongne) [reply src/rng/test/time_initkeyctr.h:32]: Great, thanks.

**Pull request merged**